huaweicloud/services/secmaster/resource_huaweicloud_secmaster_incident.go

```diff
@@ -0,0 +1,595 @@
+	}
+
+	if d.HasChanges(updateIncidentChanges...) {
+		// updateIncident: Update the configuration of SecMaster incident
+		var (
+			updateIncidentHttpUrl = "v1/{project_id}/workspaces/{workspace_id}/soc/incidents/{id}"
+			updateIncidentProduct = "secmaster"
+		)
+		updateIncidentClient, err := cfg.NewServiceClient(updateIncidentProduct, region)
+		if err != nil {
+			return diag.Errorf("error creating SecMaster Client: %s", err)
+		}
+
+		updateIncidentPath := updateIncidentClient.Endpoint + updateIncidentHttpUrl
+		updateIncidentPath = strings.ReplaceAll(updateIncidentPath, "{project_id}", updateIncidentClient.ProjectID)
+		updateIncidentPath = strings.ReplaceAll(updateIncidentPath, "{workspace_id}", fmt.Sprintf("%v", d.Get("workspace_id")))
+		updateIncidentPath = strings.ReplaceAll(updateIncidentPath, "{id}", d.Id())
+
+		updateIncidentOpt := golangsdk.RequestOpts{
+			KeepResponseBody: true,
+			OkCodes: []int{
+				200,
+			},
+		}
+
+		updateOpts, err := buildIncidentBodyParams(d, cfg)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		updateIncidentOpt.JSONBody = utils.RemoveNil(updateOpts)
+		_, err = updateIncidentClient.Request("PUT", updateIncidentPath, &updateIncidentOpt)
+		if err != nil {
+			return diag.Errorf("error updating Incident: %s", err)
+		}
+	}
+	return resourceIncidentRead(ctx, d, meta)
+}
+
+func resourceIncidentDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	workspaceID := d.Get("workspace_id").(string)
+
+	// deleteIncident: Delete an existing SecMaster incident
+	var (
+		deleteIncidentHttpUrl = "v1/{project_id}/workspaces/{workspace_id}/soc/incidents"
+		deleteIncidentProduct = "secmaster"
+	)
+	deleteIncidentClient, err := cfg.NewServiceClient(deleteIncidentProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating SecMaster Client: %s", err)
+	}
+
+	deleteIncidentPath := deleteIncidentClient.Endpoint + deleteIncidentHttpUrl
+	deleteIncidentPath = strings.ReplaceAll(deleteIncidentPath, "{project_id}", deleteIncidentClient.ProjectID)
+	deleteIncidentPath = strings.ReplaceAll(deleteIncidentPath, "{workspace_id}", fmt.Sprintf("%v", workspaceID))
+
+	deleteIncidentOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	deleteIncidentOpt.JSONBody = utils.RemoveNil(buildDeleteIncidentBodyParams(d))
+	_, err = deleteIncidentClient.Request("DELETE", deleteIncidentPath, &deleteIncidentOpt)
+	if err != nil {
+		return diag.Errorf("error deleting Incident: %s", err)
+	}
+
+	return nil
+}
+
+func buildDeleteIncidentBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"batch_ids": utils.ValueIngoreEmpty(d.Id()),
+	}
+	return bodyParams
+}
+
+func resourceIncidentImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.SplitN(d.Id(), "/", 2)
```

Comment:
it's better to parse the import string with Split instead of SplitN
```suggestion
	parts := strings.Split(d.Id(), "/", 2)
```

Reply:
fixed